framework/web_app/src/pages/StockOverview.vue

```diff
@@ -1,96 +1,260 @@
+                                />
+                            </q-item-section>
+                            <q-item-section>
+                                <q-item-label>
+                                    {{ scope.opt.description }}
+                                </q-item-label>
+                            </q-item-section>
+                        </q-item>
+                    </template>
+                </q-select>
+
+                <!-- TODO: IMPLEMENT -->
+                <q-select
+                    label="Stock Group"
+                    v-model="createStockItemForm.stock_group_id"
+                ></q-select>
+
+                <!-- TODO: IMPLEMENT -->
+                <q-select
+                    label="Stock Location"
+                    v-model="createStockItemForm.stock_location_id"
+                ></q-select>
+
+                <q-card-actions align="right">
+                    <q-btn
+                        type="submit"
+                        align="right"
+                        color="cyan"
+                        label="Save & Continue"
+                        size="lg"
+                    />
+                    <q-btn
+                        type="submit"
+                        align="right"
+                        color="green"
+                        label="Save & Close"
+                        size="lg"
+                        v-close-popup
+                    />
+                </q-card-actions>
+            </q-form>
+        </q-card>
+    </q-dialog>
+
+    <q-dialog v-model="shouldDisplayAddToShoppingCartModal">
+        <q-card style="width: 700px; max-width: 80vw">
+            <q-card-section class="row items-center">
+                <span class="q-ml-sm">
+                    Add {{ 'stock item' }} to a shopping list.
+                </span>
+            </q-card-section>
+
+            <q-card-actions align="right">
+                <q-btn
+                    color="primary"
+                    flat
+                    label="Cancel"
+                    v-close-popup
+                />
+                <q-btn
+                    color="primary"
+                    flat
+                    label="Add to list"
+                    v-close-popup
+                />
+            </q-card-actions>
+        </q-card>
+    </q-dialog>
 </template>
+
+<script lang="ts" setup>
+    import { StockLevel } from 'src/models/StockLevel';
+    import { CreateStockItemCommand } from 'src/services/api/StockItemApiService';
+    import { useStockItemStore } from 'src/stores/stockItemStore';
+    import { useStockLevelStore } from 'src/stores/stockLevelStore';
+    import { Ref, ref } from 'vue';
+
+    // region Common
+
+    const stockItemStore = useStockItemStore();
+    const stockLevelStore = useStockLevelStore();
+
+    const { createStockItemAsync, updateStockLevelAsync, stockItems } =
+        stockItemStore;
+    const { getStockLevelColour, stockLevels } = stockLevelStore;
```

Comment:
Just learning this myself. You are losing reactivity on stockItems and stockLevels destructuring like that. Read up on storeToRefs. Basically looks like destructure twice, once for reactive stuff with storeatoRefs and once for actions.

https://pinia.vuejs.org/core-concepts/

![image](https://github.com/BenTalese/DiscountDora/assets/95552720/1ac6c334-85e8-4c80-9974-7c0dc7c76e60)

Reply:
d